iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -234,26 +234,44 @@ Viewer.prototype.contents = function() {
     return this._iframes.contents();
 }
 
-
-Viewer.prototype._selectAdjacentTag = function (offset) {
-    var elements = $(".ixbrl-element:not(.ixbrl-continuation)", this._contents);
-    var current = $(".ixbrl-selected:not(.ixbrl-continuation)", this._contents);
-    var next;
-    if (current.length == 1) {
-        next = elements.eq((elements.index(current.first()) + offset) % elements.length);
+// Move by offset (+1 or -1) through the tags in the document in document
+// order.
+//
+// Each element may have one or more tags associated with it, so we need to
+// move through the list of tags associated with the current element before
+// moving to the next/prev element
+//
+Viewer.prototype._selectAdjacentTag = function (offset, currentFact) {
+    const elements = $(".ixbrl-element:not(.ixbrl-continuation)", this._contents);
+    const currentElement = $(".ixbrl-selected:not(.ixbrl-continuation)", this._contents);
+    var nextId;
+
+    if (currentElement.length == 1 && currentFact !== undefined) {
```

Comment:
Out of curiosity, what is the relevance of the `currentElement.length == 1` check?  It seems like the logic would just be "if there's a selected item, move to the prev/next selected item", but perhaps I'm overlooking some case.

Reply:
I think the two parts should be redundant, as even where we have a hidden fact, the element for it in the hidden section gets the `xbrl-selected` class added, but as the contained code will behave unpredictably if either condition is not met it seems reasonable to explicitly check both.  I'll add a comment to the effect that the check is hopefully redundant.